Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.AzureAD.UI;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Identity.Web.TokenCacheProviders.Distributed
+{
+    /// <summary>
+    /// An implementation of token cache for both Confidential and Public clients backed by MemoryCache.
+    /// </summary>
+    /// <seealso cref="https://aka.ms/msal-net-token-cache-serialization"/>
+    public class MsalDistributedTokenCacheProvider : MsalAbstractTokenCacheProvider
```

Comment:
Personally I would call these adapters. Terminology is hard here because the noun `cache` is overused. The MSAL caches have nothing to do with the IDistribuitedCache asp abstraction. I would call it `DistribuitedCacheAdapter` or smth similar.

Reply:
Yes, I'm not happy with the name **Provider**. I was thinking of **Serializer**, but in the case of this one, this is indeed, and adapter. 
Let's discuss as a team.

Reply:
Done for your suggestion, @bgavrilMS